rust/examples/recordbatch-writer.rs

```diff
@@ -0,0 +1,248 @@
+/*
+ * This file is a self-contained example of writing to a Delta table using the Arrow
+ * `RecordBatch` API rather than pushing the data through a JSON intermediary
+ *
+ *
+ * This example was originally posted by @rtyler in:
+ *      <https://github.com/buoyant-data/demo-recordbatch-writer>
+ */
+
+use chrono::prelude::*;
+use deltalake::action::*;
+use deltalake::arrow::array::*;
+use deltalake::arrow::record_batch::RecordBatch;
+use deltalake::writer::{DeltaWriter, RecordBatchWriter};
+use deltalake::*;
+use log::*;
+
+use std::collections::HashMap;
+use std::path::Path;
```

Comment:
Object store paths may work better with our APIs. For example, they implement `AsRef<str>`, so we don't need to call `.to_str().expect()` like you do below.

```suggestion
use object_store::path::Path;
```